sympy/sets/sets.py

```diff
@@ -534,6 +535,70 @@ def measure(self):
         """
         return self._measure
 
+    @property
+    def kind(self):
+        """
+        This is a method for Sets class and its subclasses such as FiniteSet,
+        EmptySet, etc.
+
+        Every instance of Set or its subclass will have kind of class
+        ``SetKind``.
+        Parameter of this SetKind will depend on arguments of that set
+
+        Parameter of SetKind can be ``NumberKind``, ``UndefinedKind``,
+        ``TupleKind`` etc.
+        if all arguments of expression are not same then parameter will be
+        ``UndefinedKind``
```

Comment:
It would be better to say that if the elements of a set do not all have the same kind then the kind will be `SetKind(UndefinedKind)`.